vtysh/vtysh.c

```diff
@@ -2662,19 +2662,17 @@ DEFUNSH(VTYSH_KEYS, vtysh_quit_keys, vtysh_quit_keys_cmd, "quit",
 	return vtysh_exit_keys(self, vty, argc, argv);
 }
 
-#if defined(HAVE_PATHD)
-DEFUNSH(VTYSH_PATHD, vtysh_exit_pathd, vtysh_exit_pathd_cmd, "exit",
+DEFUNSH(VTYSH_SR, vtysh_exit_pathd, vtysh_exit_pathd_cmd, "exit",
```

Comment:
Still not right. We can't use this function as an exit function for VTYSH_SR daemons.
This function is used as an exit function in a lot of other pathd-specific nodes.
Imagine the following config:
```
nfware(config)# segment-routing
nfware(config-sr)# traffic-eng
nfware(config-sr-te)#
```

SR-TE node is pathd-only, so when we issue "exit" in this node, we have to send it only to pathd.

We have to introduce separate `vtysh_exit_segment_routing` function with VTYSH_SR daemons list for this.

Reply:
fixed